### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by bgamari**
Fix a variety of papercuts I recently encountered.

### CODE REVIEW

File: Data/Primitive/Addr.hs [hunk 1]
```diff
@@ -100,3 +107,11 @@ setAddr :: (Prim a, PrimMonad m) => Addr -> Int -> a -> m ()
 {-# INLINE setAddr #-}
 setAddr (Addr addr#) (I# n#) x = primitive_ (setOffAddr# addr# 0# n# x)
 
+-- | Convert an 'Addr' to an 'Int'.
+addrToInt :: Addr -> Int
+{-# INLINE addrToInt #-}
+addrToInt (Addr addr#) = I# (addr2Int# addr#)
```
Comment (cartazio) [inline Data/Primitive/Addr.hs]: wouldn't Word be more correct / robustly portable?
(granted both options are sad and very very very ghc RTS/semantics specific, so i can't complain too much :) )

File: Data/Primitive/Addr.hs [hunk 2]
```diff
@@ -100,3 +107,11 @@ setAddr :: (Prim a, PrimMonad m) => Addr -> Int -> a -> m ()
...
+
+-- | Produce a string representation of an 'Addr'. Useful while debugging.
+showAddr :: Addr -> String
+showAddr addr = "0x" ++ showHex (addrToInt addr) ""
```
Comment (cartazio) [inline Data/Primitive/Addr.hs]: what happens on systems where this looks like a negative number?

File: Data/Primitive/ByteArray.hs [hunk 3]
```diff
@@ -70,10 +70,13 @@ newPinnedByteArray (I# n#)
...
--- | Create a /pinned/ byte array of the specified size and with the give
+-- | Create a /pinned/ byte array of the specified size and with the given
 -- alignment. The garbage collector is guaranteed not to move it.
 newAlignedPinnedByteArray
-  :: PrimMonad m => Int -> Int -> m (MutableByteArray (PrimState m))
+  :: PrimMonad m
+  => Int  -- ^ size
+  -> Int  -- ^ alignment
+  -> m (MutableByteArray (PrimState m))
 {-# INLINE newAlignedPinnedByteArray #-}
```
Comment (cartazio) [inline Data/Primitive/ByteArray.hs]: good clarification and seems correct :)

File: Data/Primitive/Addr.hs (hunk 1 above)
Comment (bgamari) [inline Data/Primitive/Addr.hs]: Indeed it would. Fixing.

Comment (bgamari) [inline Data/Primitive/Addr.hs]: Actually, never mind; I think given that GHC exposes `addr2Int#` and not `addr2Word#` I'd rather follow the prior art.

File: Data/Primitive/Addr.hs (hunk 2 above)
Comment (bgamari) [inline Data/Primitive/Addr.hs]: Fixing.

**Comment by bgamari** [comment]
I'm going to close this and open another from another branch of my fork.

**Pull request closed without merging**